lti_consumer/plugin/views.py

```diff
@@ -117,41 +126,200 @@ def public_keyset_endpoint(request, usage_id=None):
         response = JsonResponse(lti_config.lti_1p3_public_jwk)
         response['Content-Disposition'] = 'attachment; filename=keyset.json'
         return response
-    except (LtiError, InvalidKeyError, ObjectDoesNotExist) as exc:
+    except (LtiError, InvalidKeyError, ObjectDoesNotExist, ValueError) as exc:
         log.info("Error while retrieving keyset for usage_id %r: %s", usage_id, exc)
         raise Http404 from exc
 
 
 @require_http_methods(["GET", "POST"])
-def launch_gate_endpoint(request, suffix=None):
+def launch_gate_endpoint(request, suffix=None):  # pylint: disable=unused-argument
     """
     Gate endpoint that triggers LTI launch endpoint XBlock handler
 
-    This is basically a passthrough function that uses the
-    OIDC response parameter `login_hint` to locate the block
-    and run the proper handler.
+    This uses the location key from the "login_hint" query parameter
+    to identify the LtiConfiguration and its consumer to generate the
+    LTI 1.3 Launch Form.
     """
+    usage_id = request.GET.get('login_hint')
+    if not usage_id:
+        return render(request, 'html/lti_1p3_launch_error.html', status=400)
+
     try:
-        usage_key_str = request.GET.get('login_hint')
-        usage_key = UsageKey.from_string(usage_key_str)
+        usage_key = UsageKey.from_string(usage_id)
+    except InvalidKeyError:
+        return render(request, 'html/lti_1p3_launch_error.html', status=404)
```

Comment:
Nit: let's use `HTTP_404_NOT_FOUND`. For consistency, we could convert all status codes to constants within this diff.

Reply:
Updated all instances where integers were used.